cogs/notification.py

```diff
@@ -97,10 +98,12 @@ async def notifier(self, itn : discord.Interaction, username: str, channel: disc
         conn.row_factory = sqlite3.Row
         cursor = conn.cursor()
         
-        cursor.execute('SELECT user_id FROM notification, user WHERE username = ? AND channel_id = ? AND user_id = id', (username, str(channel.id)))
+        server_id = str(channel.guild.id)
+        
+        cursor.execute('SELECT user_id FROM notification, user WHERE username = ? AND channel_id = ? AND user_id = id AND server_id = ?', (username, str(channel.id), server_id))
```

Comment:
{user_id, channel_id} is the primary key of notification table. They are already non-repeatable, so there is no need to add server_id to the condition.